## Тема: Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона.

In [12]:
import numpy as np
from math import factorial

### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Можно применить формулу Бернулли $P_n = C_n^k p^k q^{n-k}$  

In [14]:
# функция расчета количества сочетаний
def combinations(k, n):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [42]:
n = 100
k = 85
p = 0.8
q = 1 - p
P_n = combinations(k, n) * (p**k) * (q**(n-k))
P_n

0.048061793700746355

**Ответ:** Вероятность того, что стрелок попадет в цель ровно 85 раз равно 0.048

### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.   

### Какова вероятность, что ни одна из них не перегорит в первый день? 

Можно применить формулу Бернулли Пуассона $P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$

In [4]:
m1 = 0
n = 5000
p = 0.0004
lambda_ = n * p

P1 = (lambda_**m1 / np.math.factorial(m1)) * np.exp(-lambda_)
P1

0.1353352832366127

### Какова вероятность, что перегорят ровно две? 

In [6]:
m2 = 2
n = 5000
p = 0.0004
lambda_ = n * p

P2 = (lambda_**m2 / np.math.factorial(m2)) * np.exp(-lambda_)
P2

0.2706705664732254

**Ответ:** вероятность, что ни одна лампочка не перегорит равна 0,1353 или 13%
           вероятность, что перегорят ровно две лампочки равна 0,2706 или 27%

### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?  

Можно применить формулу классической вероятности $P(A) = m / n$  
m = 70  
n = 144

In [1]:
P = 70/144
P

0.4861111111111111

**Ответ:** вероятность попадания равна 0.4861 или 49%

### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. 

### Какова вероятность того, что все мячи белые?

In [48]:
# вероятность извлечения 2 белых мячей из 1 и 2 ящика
С = 7/10 * 6/9 * 9/11 * 8/10
С

0.3054545454545454

### Какова вероятность того, что ровно два мяча белые?

### Какова вероятность того, что хотя бы один мяч белый?